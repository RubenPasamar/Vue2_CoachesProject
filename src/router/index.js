import Vue from "vue";
import VueRouter from "vue-router";

import CoachesList from "@/views/coaches/CoachesList.vue";
import NotFoundPage from "@/views/NotFoundView.vue";

Vue.use(VueRouter);

const routes = [
  { path: "/", name: "home", redirect: "/coaches" },
  { path: "/coaches", name: "coaches", component: CoachesList },
  {
    path: "/coaches/:id",
    name: "coach",
    component: () =>
      import(
        /*webpackChunkName: "coach-details"*/ "@/views/coaches/CoachDetails.vue"
      ),
    props: true,
    children: [
      {
        path: "contact",
        name: "contact",
        component: () =>
          import(
            /*webpackChunkName: "coach-contact"*/ "@/views/requests/ContactCoach.vue"
          ),
      },
    ],
  },
  {
    path: "/register",
    name: "register",
    component: () =>
      import(
        /*webpackChunkName: "coach-registration"*/ "@/views/coaches/CoachRegistration.vue"
      ),
  },
  {
    path: "/requests",
    name: "requests",
    component: () =>
      import(
        /*webpackChunkName: "requests-received"*/ "@/views/requests/RequestsReceived.vue"
      ),
  },
  { path: "/:notFound(.*)", component: NotFoundPage },
];

const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  routes,
});

export default router;
