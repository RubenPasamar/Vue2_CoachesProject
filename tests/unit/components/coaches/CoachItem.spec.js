import { shallowMount, createLocalVue } from "@vue/test-utils";
import CoachItem from "@/components/coaches/CoachItem.vue";
import BaseButton from "@/components/UI/BaseButton.vue";
import BaseBadge from "@/components/UI/BaseBadge.vue";
import router from "@/router";
import VueRouter from "vue-router";

const item = {
  id: "c1",
  firstName: "Maximilian",
  lastName: "Schwarzmüller",
  areas: ["frontend", "backend", "career"],
  description:
    "I'm Maximilian and I've worked as a freelance web developer for years. Let me help you become a developer as well!",
  hourlyRate: 30,
};

const localVue = createLocalVue();
localVue.use(VueRouter);

describe("CoachItem.vue", () => {
  const wrapper = shallowMount(CoachItem, {
    localVue,
    router,
    propsData: {
      id: item.id,
      firstName: item.firstName,
      lastName: item.lastName,
      areas: item.areas,
      description: item.description,
      rate: item.hourlyRate,
    },
    components: {
      BaseButton: BaseButton,
      BaseBadge: BaseBadge,
    },
    computed: {
      fullName: () => {
        return "Maximilian Schwarzmüller";
      },
    },
  });

  it("Should match Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Should have same base-badge as areas", () => {
    const baseBadges = wrapper.findAll("base-badge-stub");
    expect(baseBadges.length).toEqual(item.areas.length);
  });

  it("Should have 2 base-buttons", () => {
    const baseButtons = wrapper.findAll("base-button-stub");
    expect(baseButtons.length).toBe(2);
  });

  it("Should not have text base-badge's components", () => {
    const baseBadge = wrapper.findAll("base-badge-stub");
    for (let i = 0; i < baseBadge.length; i++) {
      expect(baseBadge.at(i).text()).toMatch("");
    }
  });

  it("Should have text base-button components", () => {
    const baseBadge = wrapper.findAll("base-badge-stub");
    for (let i = 0; i < baseBadge.length; i++) {
      expect(baseBadge.at(i).text()).toMatch("");
    }
  });

  it("render hourlyRate", () => {
    expect(wrapper.find("h4").text()).toBe("$30/hour");
  });
});
