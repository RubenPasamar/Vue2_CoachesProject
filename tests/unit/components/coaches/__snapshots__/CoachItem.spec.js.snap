// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoachItem.vue Should match Snapshot 1`] = `
<li description="I'm Maximilian and I've worked as a freelance web developer for years. Let me help you become a developer as well!">
  <h3>Maximilian Schwarzm√ºller</h3>
  <h4>$30/hour</h4>
  <div>
    <base-badge-stub title="frontend" type="frontend"></base-badge-stub>
    <base-badge-stub title="backend" type="backend"></base-badge-stub>
    <base-badge-stub title="career" type="career"></base-badge-stub>
  </div>
  <div class="actions">
    <base-button-stub mode="outline" link="true" to="/coaches/c1/contact">Contact</base-button-stub>
    <base-button-stub link="true" to="/coaches/c1">View Details</base-button-stub>
  </div>
</li>
`;
